### PR TITLE
Implementation of Head Tracking metrics

### PR DESCRIPTION
**Pull request by Sentient07**
Hello Jonathon,

I've implemented the metric reported in [our paper ](https://arxiv.org/pdf/2103.13516.pdf). I verified and am able to reproduce the results in Table 12 with this branch. Of course the first draft isn't perfect, but will be happy to make modifications to better adapt to the design of this repository.

Regards,

### CODE REVIEW

File: trackeval/datasets/mot_challenge_2d_box.py [hunk 1]
```diff
@@ -323,8 +327,9 @@ def get_preprocessed_seq_data(self, raw_data, cls):
         self._check_unique_ids(raw_data)
 
         distractor_class_names = ['person_on_vehicle', 'static_person', 'distractor', 'reflection']
-        if self.benchmark == 'MOT20':
+        if self.benchmark == 'MOT20' and self.benchmark == 'HT':
```
Comment (JonathonLuiten) [inline trackeval/datasets/mot_challenge_2d_box.py:330]: These seems like and error. I think you mean OR here??

**Comment by JonathonLuiten** [comment]
Okay I've looked through the code. In general I would like it if you refactor this such that it is'disentangled' from the original MOTChallege dataset AND IDF1 metric.

e.g. instead of editing the motchallenge dataset class, please create a head tracking class (how you name it up to you, by try make it consistent with what is there).

And instead of editing IDF1, please create a new euclidian IDF1 class.

Of course for both of these you can copy the code and only adjust minimally, but let's have these as separate things to not add dependencies and complications to the originals.

In general this means there will be a bit of duplicated code (not ideal), but overall the code will be cleaner, easier to read and understand, and easier to maintain and keep bug free.

Cheers,
Jono

File: trackeval/datasets/mot_challenge_2d_box.py (hunk 1 above)
Comment (Sentient07) [reply trackeval/datasets/mot_challenge_2d_box.py:330]: Oops, yes, it should've been an OR

**Comment by Sentient07** [comment]
Hi,

Thanks for your comments. Separating out makes sense and I will do it by the end of this week. Main idea for juxtaposing is to better elucidate the differences.

**Comment by JonathonLuiten** [comment]
The error you are getting is because when you combine the results from all 'sequences' into an overall result, this overall result no longer contains the IDEulcid metric.

E.g. there is the function 'combine_sequences' which first combines integege values and then calls _compute_final_fields to combine the other value. But for you this never calculates the combined IDeuclid metric. 

You need to add code here to combine IDeuclid over all seqeunce. ( you can see how this is done for IDF1 for example).

Jono

**Comment by JonathonLuiten** [comment]
One more thing. If you look at 'files changed' you've changed and added files that don't need to be changed. E.g. change clear and added some .clear binaries. Please clean this up.

**Comment by Sentient07** [comment]
Done! :)

**Pull request merged**